misc/pragmas.md

```diff
@@ -0,0 +1,550 @@
+namespace, some into another, and some into none, and that namespace
+bindings should remain constant throughout the grammar (so: no
+changing the default namespace in the middle of the document).
+
+We define the namespace 'http://example.com/ixml-namespaces" as
+providing for namespace bindings, and we adopt the convention that a
+prefix `ppp` is bound to a namespace name `nnn` if a pragma of the form
+````
+    [@nsd:ppp nnn]
+````
+
+is encountered in a context where the prefix *nsd* is bound to
+"'http://example.com/ixml-namespaces".
+
+To bootstrap the process, we adopt the principle that a 
+pragma of the form
+````
+    [@ns:ns http://example.com/ixml-namespaces]
+````
+
+binds the prefix `ns` to that namespace. (Alternatively, we could
+follow the example of XML namespaces and reserve some string for
+namespace bindings, but this example is trying to show a way to do
+this that does not rely on a decree in the spec but could in principle
+be handled as an extension.)
+
+So a grammar in which some elements are in the *x* namespace, some in
+the *y* namespace, and some in no namespace at all, might be
+
+````
+    [@nsd:nsd http://example.com/ixml-namespaces]
+    [@nsd:x http://example.org/NS/existential]
+    [@nsd:y http://example.com/NS/yoyo]
+    x:sentence: x:a, ' ', y:b?, '.  ', c.
+	x:a: 'Speed'.
+	y:b: 'kills'.
+	c:  'It really does.'.
+````
+The XML representation of the grammar might be:
+
+````
+    <ixml>
+      <rule name="x:sentence"
+        nsd:nsd="http://example.com/ixml-namespaces"
+        nsd:x="http://example.org/NS/existential"
+        nsd:y="http://example.com/NS/yoyo"
+        >
+        <alt>
+          <nonterminal name="x:a"/>
+          <literal sstring=" "/>
+          <option>
+            <nonterminal name="y:b"/>
+          </option>
+          <literal sstring=".  "/>
+          <nonterminal name="c"/>
+        </alt>
+      </rule>
+      <rule name="x:a">
+        <alt>
+          <literal sstring="Speed"/>
+        </alt>
+      </rule>
+      <rule name="y:b">
+        <alt>
+          <literal sstring="kills"/>
+          </alt>
+      </rule>
+      <rule name="c">
+        <alt>
+          <literal sstring="It really does."/>
+        </alt> 
+      </rule>
+    </ixml>
+````
+
+*We have another bootstrapping problem here: because the prefix*
+`nsd` *is not declared, the XML shown is not namespace-well-formed. We
+need some way of getting it declared, preferably other than by fiat,
+but I don't currently see an alternative to fiat.*
+
+To make the XML be namespace well-formed, we will need somehow to get
+at least a namespace declaration for *nsd* to be injected on the
+`ixml` element, and preferably also namespace bindings for the
+prefixes *x* and *y*, so the `ixml` start tag takes something like the
+following form.
+````
+    <ixml xmlns:nsd="http://example.com/ixml-namespaces"
+          xmlns:x="http://example.org/NS/existential"
+          xmlns:y="http://example.com/NS/yoyo" >
+ ````
```

Comment:
I think we should specify this as the desired output instead of the opening tag in the code example above, see my previous comment.